modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -2490,6 +2490,44 @@ private void createNewStruct(int[] operands, StackFrame sf) {
         StructureRefCPEntry structureRefCPEntry = (StructureRefCPEntry) constPool[cpIndex];
         StructInfo structInfo = (StructInfo) structureRefCPEntry.getStructureTypeInfo();
         BStruct bStruct = new BStruct(structInfo.getType());
+
+        // Populate default values
+        int longRegIndex = -1;
+        int doubleRegIndex = -1;
+        int stringRegIndex = -1;
+        int booleanRegIndex = -1;
+        for (StructFieldInfo fieldInfo : structInfo.getFieldInfoEntries()) {
+            DefaultValueAttributeInfo defaultValueInfo =
+                    (DefaultValueAttributeInfo) fieldInfo.getAttributeInfo(AttributeInfo.Kind.DEFAULT_VALUE_ATTRIBUTE);
```

Comment:
Shall we do the if (defaultValueInfo != null) here and 'continue' the loop, so you don't have to repeat that in the case statements.

Reply:
But we have to increment the counter.

Reply:
We can't continue at the top, since we need to increment the corresponding regIndex, regardless of the default value.